pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -134,6 +136,7 @@ func newManagerImpl(socketPath string) (*ManagerImpl, error) {
 		return nil, fmt.Errorf("failed to initialize checkpoint manager: %v", err)
 	}
 	manager.checkpointManager = checkpointManager
+	manager.writeCheckpoint()
```

Comment:
It doesn't seem correct to write the checkpoint upon devicemanager creation. 

This was meant to help in indicating whether or not the node has been recreated. However, if we write checkpoint at creation, the checkpoint will be present before ShouldResetExtendedResourceCapacity() is called, even after node recreation.

Reply:
It looks like there won't be devices registered until genericDeviceUpdateCallback is called, which will also write the initial checkpoint. https://github.com/kubernetes/kubernetes/blob/e6413d20427cb78c20e3e3d963682a9413244db3/pkg/kubelet/cm/devicemanager/endpoint.go#L118

In the callback, it seems like the checkpoint writing (https://github.com/kubernetes/kubernetes/blob/20cd56e6cae5ad9fadca4d281ddc945c9abc01ac/pkg/kubelet/cm/devicemanager/manager.go#L156 would need the unlock after writing the checkpoint and ShouldResetExtendedResourceCapacity would also need to acquire the mutex before checking the checkpoint?